config/config.go

```diff
@@ -0,0 +1,36 @@
+package config
+
+import (
+	"io/ioutil"
+
+	"github.com/go-yaml/yaml"
+	"github.com/jinzhu/configor"
+	"github.com/mcuadros/go-defaults"
+)
+
+// App config struct
+type Config struct {
+	Endpoint string `default:"" json:"endpoint" form:"endpoint" query:"endpoint" required:"true"`
+	Username string `default:"" json:"username" form:"username" query:"username" required:"true"`
+	Password string `default:"" json:"password" form:"password" query:"password" required:"true"`
+}
+
+// Create config from configFile
+func NewConfig(configFile string) (*Config, error) {
+
+	config := new(Config)
+	defaults.SetDefaults(config)
+
+	configBytes, err := ioutil.ReadFile(configFile)
+	if err == nil {
+		err = yaml.Unmarshal(configBytes, &config)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if err := configor.Load(config); err != nil {
```

Comment:
Yes, I am using config interface from Open API, and it's designed to support default values (e.g. default factomd location https://api.factomd.net). Although restart script doesn't have default values for config, I decided to keep this feature as it doesn't increase complexity and readability of the code and may be used in future updates or forks, e.g.

Reply:
I meant that the configor library will parse the yaml file into `config` struct for you, instead of doing it yourself in addition to the configor check.

Reply:
f58f205cab982bec86f9ebce20b2f16ed3045ae1